### DISCUSSION TITLE
why attach singleton to a source react element?

### DISCUSSION THREAD

**Discussion by KutnerUri** [floating-ui/floating-ui#1236] [Q&A]
I'm trying out `useSingleton()` and it seems it's only relevant to the Popper instance, not the react element / props or tippy's props.

For example, I thought the html dom would be reused or at least props like `appendTo` and `theme`, but they don't seem to have any effect on the target poppers. (or I am just using it wrong? I did use the overrides param)

if so, would it not make more sense to hook the popper instance without a dom element?
```tsx
function Tooltiperr() {
  const popperInstance = usePopper({ delay: 500, modifiers: [...]});

  return <div>
    <Tippy popper={popperInstance} content="tooltip">
      awesome
    </Tippy>
  </div>
}
```

or event with react context:
```tsx
function Tooltiperr() {
  const popperInstance = usePopper({ delay: 500, modifiers: [...]});

  return <div>
    <PopperProvider topic='awesomes' value={popperInstance}>
        <Tippy topic='awesomes' content="tooltip">
          awesome
        </Tippy>
    </PopperProvider>
  </div>
}
```

**Comment by FezVrasta** [comment]
👋 this community is about Popper, not Tippy. Take a look at their repository if you need help.
